giraffe/src/components/geo/LayerSwitcher.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {FunctionComponent} from 'react'
+import {GeoViewLayer} from '../../types/geo'
+
+//components
+import CircleMarkerLayer from './CircleMarkerLayer'
+import HeatmapLayer from './HeatmapLayer'
+import PointMapLayer from './PointMapLayer'
+import TrackMapLayer from './TrackMapLayer'
+
+//types
+import {
+  GeoCircleViewLayer,
+  GeoHeatMapViewLayer,
+  GeoPointMapViewLayer,
+  GeoTrackMapViewLayer,
+} from '../../'
+import {GeoTable} from './processing/GeoTable'
+import {Config} from '../../types'
+
+interface Props {
+  layer: GeoViewLayer
+  preprocessedTable: GeoTable
+  index: number
+  stylingConfig: Partial<Config>
+}
+
+export const LayerSwitcher: FunctionComponent<Props> = ({
+  layer,
+  preprocessedTable,
+  index,
+  stylingConfig,
+}) => {
+  switch (layer.type) {
```

Comment:
question: is a `default` necessary here? if it can't fail here, a comment saying our code can't reach this is a good idea.

since this is a library, we may need to guard against this anyway in the future